app/ui-react/packages/api/src/useConnection.tsx

```diff
@@ -1,8 +1,16 @@
-import { ConnectionOverview } from '@syndesis/models';
-import { getActionsWithFrom, getActionsWithTo, isDerived } from './helpers';
+import { IConnectionOverview } from '@syndesis/models';
+import {
+  getActionsWithFrom,
+  getActionsWithTo,
+  isConfigurationRequired,
+  isDerived,
+  isTechPreview,
+} from './helpers';
 import { useApiResource } from './useApiResource';
 
-export const transformConnectionResponse = (connection: ConnectionOverview) => {
+export const transformConnectionResponse = (
+  connection: IConnectionOverview
+) => {
   return {
```

Comment:
I think here we should get the updated connector like `const connector = transformConnectorResponse(connection.connector!);` and use it after.
`isTechPreview` would become `isTechPreview: connector.isTechPreview`